common/persistence/client/factory.go

```diff
@@ -75,7 +75,7 @@ type (
 		// NewHistoryV2Store returns a new historyV2 store
 		NewHistoryV2Store() (p.HistoryStore, error)
 		// NewMetadataStore returns a new metadata store
-		NewMetadataStore() (p.MetadataStore, error)
+		NewMetadataStore() (p.DomainStore, error)
```

Comment:
Shall we also update the method name? We may need to update our internal repo though.

Reply:
yeah we should.

Reply:
Can we hold landing this diff before the new release branch is cut? We are fixing some internal issues and may need to land more diffs to master before cutting the release. Should only take a few days.

Reply:
ohh sure.